### PR TITLE
feat: table definition for renderer

### PR DESCRIPTION
**Pull request by kpranoto-aws**
*Description of changes:*
* Table definition to be used by table renderer
* Some types are fixed to use primitive instead of instanced 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui/lib/generate-view-definition/helpers/order-filter-columns.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    current = current.next;
+  }
+
+  return ordered;
+};
+
+export const orderAndFilterVisibleColumns = (columns: ColumnsMap, fields?: DataStoreModelField[]): ColumnInfo[] => {
+  let ordered: ColumnInfo[] = [];
+
+  if (fields) {
```
Comment (cwoolum) [inline packages/codegen-ui/lib/generate-view-definition/helpers/order-filter-columns.ts:105]: Need to check that fields has length here also or line 110 will fail.. Please add a test for this also.

Comment (kpranoto-aws) [reply packages/codegen-ui/lib/generate-view-definition/helpers/order-filter-columns.ts]: Added size check and test for not throwing error

**Review by cwoolum — APPROVED** [review]
LGTM

**Review by SwaySway — APPROVED** [review]
🚢

**Pull request merged**